src/Agent.Worker/Build/FileContainerServer.cs

```diff
@@ -232,7 +232,8 @@ private async Task<UploadResult> UploadAsync(IAsyncCommandContext context, int u
                 token.ThrowIfCancellationRequested();
                 try
                 {
-                    string itemPath = (_containerPath.TrimEnd('/') + "/" + fileToUpload.Remove(0, _sourceParentDirectory.Length + 1)).Replace('\\', '/');
+                    string fileName = fileToUpload.Replace(_sourceParentDirectory, string.Empty).Replace("\\", string.Empty).Replace("/", string.Empty);
```

Comment:
I'd suggest a FF for this fix

Reply:
@merlynomsft Yes, I've created a new FF -> https://dev.azure.com/mseng/AzureDevOps/_git/AzureDevOps/pullrequest/741463